server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/Tablet.java

```diff
@@ -231,7 +232,9 @@ enum CompactionState {
   private final ConcurrentHashMap<Long,List<TabletFile>> bulkImported = new ConcurrentHashMap<>();
 
   private final int logId;
-  private final boolean onDemand;
+
+  // TODO: User can change this, how does it get updated?
```

Comment:
Need to create a follow on issue before merging.

Reply:
captured this in the comment of follow-on tasks